components/funnel.js

```diff
@@ -0,0 +1,86 @@
+// funnel.js
+
+var _ = require('underscore');
+var fs = require('fs');
+
+var stateFactory = require('../src/create-state.js');
+var logger = require('../src/logger.js');
+var wrapper = require('../src/javascript-wrapper.js');
+
+module.exports = wrapper(funnel);
+
+/**
+ * Funnels a single input into the pipeline one at a time, and queues the rest
+ * Fires the next one when it receives an empty file name
```

Comment:
@param input Either a patientId, to be queued and sent to the output when it is its turn, or a _____[fill in something appropriate here]____ from a feedback connection, indicating that the previously running patientId finished and the next one can be sent to the output.  The input port should NOT be 'multi' but should be connected *both* to an upstream node that supplies the patientIds to be processed, and to a feedback node at the end of the pipeline that indicates when the pipeline has finished processing a patientId.   NOTE: This will need to be changed if non-'multi' nodes become prohibited from having multiple connections.